src/fides/api/service/connectors/sql_connector.py

```diff
@@ -510,6 +511,19 @@ def query_config(self, node: ExecutionNode) -> BigQueryQueryConfig:
         """Query wrapper corresponding to the input execution_node."""
         return BigQueryQueryConfig(node)
 
+    # Overrides SQLConnector.test_connection
+    def test_connection(self) -> Optional[ConnectionTestStatus]:
+        try:
+            bq_schema = BigQuerySchema(**self.configuration.secrets)
+            client = bq_schema.get_client()
+            all_projects = [project for project in client.list_projects()]
+            if all_projects:
+                return ConnectionTestStatus.succeeded
+            else:
+                raise ConnectionException("No datasets found with the provided credentials.")
```

Comment:
this is nice handling! just a tweak on the wording:
```suggestion
                raise ConnectionException("No bigquery projects found with the provided credentials.")
```

Reply:
fixed here https://github.com/ethyca/fides/pull/5138/commits/16d65db4c965c945833221f7d03c3fb705a31073#diff-15895ed412c26a8f5c097a7eddfa11273da29a8d999e7c342bdeb475224fff8aR534